support/xdg-base-dirs.lisp

```diff
@@ -0,0 +1,47 @@
+;; Copyright (C) 2020  Javier Olaechea
+
+;; This file is part of stumpwm.
+
+;; This program is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;; The code in this file deals with providing support for the XDG Base
+;; Directory Specification.
+
+;;; Code:
+
+(in-package #:stumpwm)
+
+(defun config-dir ()
+  (let ((config-home (or (sb-ext:posix-getenv "XDG_CONFIG_HOME")
+                         (merge-pathnames ".config/" (user-homedir-pathname)))))
+    (merge-pathnames "stumpwm/"
+                     config-home)))
+
+(defun ensure-config-dir ()
+  (let ((config-dir (config-dir)))
+    (unless (probe-file config-dir)
+      (sb-ext:run-program "mkdir" `("-m" "700" "-p" ,config-dir)))))
```

Comment:
It's not likely, but this will fail if the parent directory doesn't exist. Use [ensure-directories-exist](http://clhs.lisp.se/Body/f_ensu_1.htm) instead?

Reply:
I wasn't aware of `ensure-directories-exist`. It is certainly preferable to shelling out the mkdir, done.